### DISCUSSION TITLE
Understanding the `initial` parameter in `jax.nn.softmax`

### DISCUSSION THREAD

**Discussion by ayaka14732** [jax-ml/jax#18227] [General]
I have difficulty understanding the `initial` parameter in `jax.nn.softmax`:

```
    initial: The minimum value used to shift the input array. Must be present
      when :code:`where` is not None.
```

Is there a formula for that?

**Comment by jakevdp** [comment]
The `where` argument lets you pass a mask specifying which values should contribute to the aggregation. When you do this, there is the possibility that `where` contains all `False` values, in which case the return value is ambiguous. For this reason, you must set `initial` when you specify `where`, so that there is a non-ambiguous return value for this case.

This follows the API of numpy aggregations like `max` and `min`, which have the same requirements.
